_posts/2019-08-09-pairwise-distance.ipynb

```diff
@@ -0,0 +1,720 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import scipy.spatial.distance as spd\n",
+    "import dask.array as da"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def pairwise_cityblock_cpu(x):\n",
+    "    assert x.ndim == 2\n",
+    "    out = spd.pdist(x.T, metric='cityblock')\n",
```

Comment:
To record a point from earlier for follow up, `pdist` is likely using `float64`. Though the kernels written below are `float32`. Would be good to standardize these a bit to have a fair comparison. That said, this will likely only be a 2x change relative to what we have here (unless I'm missing things).

One option would be to write a Numba kernel for this case as well. Though after talking with @alimanfoo, it seems SciPy's implementation is a bit more performant than a naive Numba kernel. Also SciPy seems to handle C and Fortran order arrays with identical performance. So there may be some other tricks buried in the SciPy implementation that would be worth learning from.